File: CS50/Week 1/Problem Set 1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_spaces(int spaces);

void print_bricks(int bricks);

int main(void)
{
    int height;

    // Keep prompting the user to enter a valid height
    do
    {
        height = get_int("Height?: ");
    }
    while (height < 1);

    for (int row = 0; row < height; row++)
    {
        // Calculates the number of spaces for each row
        int numberOfSpaces = (height - 1) - row;

        print_spaces(numberOfSpaces);

        print_bricks(row);
    }
}

// Logic to print the number of spaces
void print_spaces(int spaces)
{
    for (int i = 0; i < spaces; i++)
    {
        printf(" ");
    }
}

// Logic to print the number of bricks
void print_bricks(int bricks)
{
    for (int i = 0; i < bricks + 1; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: CS50/Week 6/Problem Set 6/sentimental-mario-less/mario.py
# Function to print the number of spaces
def print_spaces(spaces):
    for i in range(spaces):
        print(" ", end="")


# Function to print the number of bricks
def print_bricks(bricks):
    for i in range(bricks + 1):
        print("#", end="")
    print("\n", end="")


def main():
    # Keep prompting the user to enter a valid height
    while True:
        try:
            height = int(input("Height?: "))
            if height >= 1 and height <= 8:
                break
        except ValueError:
            pass

    for row in range(height):
        # Calculates the number of spaces for each row
        number_of_spaces = (height - 1) - row

        print_spaces(number_of_spaces)

        print_bricks(row)


if __name__ == "__main__":
    main()
